pkg/sql/virtual_schema.go

```diff
@@ -574,11 +575,27 @@ func (e *virtualDefEntry) Desc() catalog.Descriptor {
 	return e.desc
 }
 
-func canQueryVirtualTable(evalCtx *eval.Context, e *virtualDefEntry) bool {
-	return !e.unimplemented ||
+func canQueryVirtualTable(p *planner, e *virtualDefEntry, tn *tree.TableName) error {
```

Comment:
This is the modified access check for virtual tables, which also bypass our normal descriptor privilege checking.